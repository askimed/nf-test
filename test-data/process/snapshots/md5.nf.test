nextflow_process {

  name "Test process xy"

  script "./process.nf"
  process "TEST_PROCESS"

  test("Should succeed because two unique snapshots") {

    when {
      process {
        """
        input[0] = "Lukas"
        """
      }
    }

    then {

      def content = [
        object1: "lukas",
        object2: 27,
        object3: [
            a: "lll",
            b: [1,2,3,4,5,6]
        ]
      ]

      assert process.success
      assert snapshot(content).match()
      assert snapshot(content).md5().match("lukas")
    }

  }

}
