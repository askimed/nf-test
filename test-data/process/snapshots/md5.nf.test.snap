{
    "Should succeed because two unique snapshots": {
        "content": [
            {
                "object1": "lukas",
                "object2": 27,
                "object3": {
                    "a": "lll",
                    "b": [
                        1,
                        2,
                        3,
                        4,
                        5,
                        6
                    ]
                }
            }
        ],
        "timestamp": "2024-01-28T11:41:29.018819"
    },
    "lukas": {
        "content": "a54d97026c5ed6e6a57ccd3a3f5a3cbc",
        "timestamp": "2024-01-28T11:41:29.025981"
    }
}